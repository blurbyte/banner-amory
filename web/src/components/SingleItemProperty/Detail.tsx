import styled, { css } from 'styled-components';

import { Colors } from '../../styles/colors';

type DetailProps = {
  big?: boolean;
  statLine?: boolean;
};

const Detail = styled.dd<DetailProps>`
  color: ${Colors.blue};
  line-height: ${props => (props.big ? 1 : 'inherit')};
  font-size: ${props => (props.big ? '1.8rem' : '1.4rem')};
  font-weight: ${props => (props.big ? 600 : 'inherit')};
  margin: 0;
  ${props =>
    props.statLine &&
    css`
      text-align: right;
    `};
`;

export default Detail;
