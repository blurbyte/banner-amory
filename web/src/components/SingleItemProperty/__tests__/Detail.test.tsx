import * as React from 'react';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';

import Detail from '../Detail';

test('renders correctly', () => {
  const component = renderer.create(<Detail />);
  expect(component.toJSON()).toMatchSnapshot();
});
