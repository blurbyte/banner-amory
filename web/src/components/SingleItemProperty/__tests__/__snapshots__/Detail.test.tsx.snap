// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  color: #3d26da;
  line-height: inherit;
  font-size: 1.4rem;
  font-weight: inherit;
  margin: 0;
}

<dd
  className="c0"
/>
`;
